docker-entrypoint.sh

```diff
@@ -267,6 +278,25 @@ docker_sql_escape_string_literal() {
 	echo "${escaped//\'/\\\'}"
 }
 
+# Creates replication user
+create_replica_user() {
+	if [[ "$MARIADB_REPLICATION_MODE" = "master" && -n "$MARIADB_REPLICATION_USER" ]]; then
+		if [ -n "$MARIADB_REPLICATION_PASSWORD" ] || [ -n "$MARIADB_REPLICATION_PASSWORD_HASH" ]; then
+			mysql_note "Creating user ${MARIADB_REPLICATION_USER}"
+			if [ -n  "$MARIADB_REPLICATION_PASSWORD_HASH" ]; then
+				createUser="CREATE USER '$MARIADB_REPLICATION_USER'@'%' IDENTIFIED BY PASSWORD '$MARIADB_REPLICATION_PASSWORD_HASH';"
+			else
+				# SQL escape the user password, \ followed by '
+				local userPasswordEscaped
+				userPasswordEscaped=$( docker_sql_escape_string_literal "${MARIADB_REPLICATION_PASSWORD}" )
+				createUser="CREATE USER '$MARIADB_REPLICATION_USER'@'%' IDENTIFIED BY '$userPasswordEscaped';"
```

Comment:
Rather than set an global variable `createUser` which hides the other local `createUser` in the calling function. Echo the `CREATE USER...` to stdout.

Reply:
Done 👍🏽